src/main/java/anodot/stage/destination/AnodotTarget.java

```diff
@@ -139,6 +147,33 @@ public void write(Batch batch) throws StageException {
         }
     }
 
+    private Record getLastRecord(Batch batch) {
+        Iterator<Record> iter = batch.getRecords();
+        Record record = null;
+        while (iter.hasNext()) {
+            record = iter.next();
+        }
+        return record;
+    }
+
+    private void sendOffsetToAgent(String offset, String pipelineId) throws Exception {
+        HttpClientCommon httpClientCommon = new HttpClientCommon(new JerseyClientConfigBean());
+        List<ConfigIssue> issues = super.init();
+        httpClientCommon.init(issues, getContext());
+        WebTarget target = httpClientCommon.getClient().target(conf.agentOffsetUrl + pipelineId);
+        Invocation.Builder builder = target.request();
+        MultivaluedMap<String, Object> requestHeaders = new MultivaluedHashMap<>();
+        String contentType = HttpStageUtil.getContentType(requestHeaders, DataFormat.JSON);
+        Response response = builder.method(String.valueOf(HttpMethod.POST), Entity.entity(String.format("{\"offset\": \"%s\"}", offset).getBytes(StandardCharsets.UTF_8), contentType));
+
+        if (response.getStatus() < 200 || response.getStatus() >= 300) {
+            String responseEntity = response.readEntity(String.class);
+            response.close();
+            throw new Exception("Failed to save agent offset, response: " + responseEntity);
+        }
+        response.close();
```

Comment:
will not be executed when the exception will be thrown before.
You can use try-finally